pkgs/development/python-modules/disnake/default.nix

```
@@ -0,0 +1,68 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+  setuptools,
+
+  aiohttp,
+
+  withVoice ? true,
+  pynacl,
+  libopus,
+  ffmpeg,
+}:
+
+buildPythonPackage rec {
+  pname = "disnake";
+  version = "2.9.2";
+
+  pyproject = true;
+  build-system = [ setuptools ];
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "DisnakeDev";
+    repo = "disnake";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-J7nNVHhkQ9ekGjioObxaTL0OhjLjHHYlgbPdAG9Vz4o=";
+  };
+
+  dependencies =
+    [ aiohttp ]
+    ++ lib.optionals withVoice [
+      pynacl
+      libopus
+      ffmpeg
+    ];
+
+  postPatch = lib.optionalString withVoice ''
+    substituteInPlace "disnake/opus.py" \
+      --replace "ctypes.util.find_library('opus')" "'${libopus}/lib/libopus.so.0'"
+    substituteInPlace "disnake/player.py" \
+      --replace "executable='ffmpeg'" "executable='${ffmpeg}/bin/ffmpeg'"
```

I think you want `--replace-fail` -- `--replace` just does nothing if it can't find what it's looking for.

```suggestion
      --replace-fail "ctypes.util.find_library('opus')" "'${libopus}/lib/libopus.so.0'"
    substituteInPlace "disnake/player.py" \
      --replace-fail "executable='ffmpeg'" "executable='${ffmpeg}/bin/ffmpeg'"
```

---

This should probably edited in the original discord.py as well.

---

Yep. https://github.com/NixOS/nixpkgs/issues/21195, https://github.com/NixOS/nixpkgs/pull/260776